evm/src/NttManager/NttManagerState.sol

```diff
@@ -251,29 +255,35 @@ abstract contract NttManagerState is
     }
 
     /// @inheritdoc INttManagerState
-    function setPeer(uint16 peerChainId, bytes32 peerContract) public onlyOwner {
+    function setPeer(uint16 peerChainId, bytes32 peerContract, uint8 decimals) public onlyOwner {
         if (peerChainId == 0) {
             revert InvalidPeerChainIdZero();
         }
         if (peerContract == bytes32(0)) {
             revert InvalidPeerZeroAddress();
         }
+        if (decimals == 0) {
+            revert InvalidPeerDecimals();
+        }
 
-        bytes32 oldPeerContract = _getPeersStorage()[peerChainId];
+        NttManagerPeer memory oldPeer = _getPeersStorage()[peerChainId];
 
-        _getPeersStorage()[peerChainId] = peerContract;
+        _getPeersStorage()[peerChainId].peerAddress = peerContract;
+        _getPeersStorage()[peerChainId].tokenDecimals = decimals;
```

Comment:
Maybe we should add a check here that if the previous token decimals were non-zero then the new token decimals should be the same?

I don't think this is a security issue since we check the manager that's encoded in the payload (specifically thinking about in-flight transfers) but just intrigued what you think

Reply:
I think we can just keep it this way, good flag though